store/index.js

```diff
@@ -191,9 +196,21 @@ export const actions = {
       await dispatch('reset', activeNetwork);
     }
 
+    const loadFromKetherView = activeNetwork == 'rinkeby'; // Only deployed KetherView on rinkeby
     const loadFromEvents = true; // Okay to do it always? or should we do: state.loadedBlockNumber > 0
 
-    if (loadFromEvents) {
+    if (loadFromKetherView) {
+      const limit = 200;
+      const len = await contract.getAdsLength();
+      commit('setAdsLength', len);
+      //TODO test off-by-ones
+      for (let offset = 0; offset < len; offset+=limit) {
+        const ads = await ketherView.allAds(contract.address, ketherNFT.address, offset, limit);
```

Comment:
We can do these concurrently speed things up (accumulate without await and `Promise.all` to merge promises?)